app/services/shopify_import/variants/data_parsers/base_data.rb

```diff
@@ -0,0 +1,49 @@
+module ShopifyImport
+  module Variants
+    module DataParsers
+      class BaseData
+        def initialize(shopify_variant, spree_product)
+          @shopify_variant = shopify_variant
+          @spree_product = spree_product
+        end
+
+        def variant_attributes
+          {
+            sku: @shopify_variant.sku,
+            price: @shopify_variant.price,
+            weight: @shopify_variant.grams,
+            position: @shopify_variant.position,
+            product_id: @spree_product.id
+          }
+        end
+
+        def option_value_ids
+          %w[option1 option2 option3].map do |option_name|
+            next unless (option_value = @shopify_variant.send(option_name))
+
+            Spree::OptionValue.find_by!(
+              option_type_id: @spree_product.option_type_ids,
+              name: option_value.strip.downcase
+            ).id
+          end
+        end
+
+        def track_inventory
+          @shopify_variant.inventory_management.eql?('shopify')
+        end
+
+        def backorderable
+          @shopify_variant.inventory_policy.eql?('continue')
+        end
+
+        def inventory_quantity
+          @shopify_variant.inventory_quantity
+        end
+
+        def stock_location
+          Spree::StockLocation.where(name: I18n.t(:shopify)).first_or_create(default: false, active: false)
```

Comment:
Please use `first_or_create_by` instead of `first_or_create` as it's been "silently deprecated":
https://github.com/mdeering/attribute_normalizer/issues/68#issuecomment-73005120